relay.go

```diff
@@ -654,6 +665,61 @@ func (r *Relayer) handleLocalCallReq(cr *lazyCallReq) bool {
 	return true
 }
 
+func (r *Relayer) fragmentingSend(f *lazyCallReq, relayToDest relayItem, origID uint32) error {
+	// TODO(echung): should we pool the writers?
+	fragWriter := newFragmentingWriter(
+		r.logger, r.newFragmentSender(relayToDest.destination, f, origID),
+		f.checksumType.New(),
+	)
+	defer r.conn.opts.FramePool.Release(f.Frame)
+
+	arg2Writer, err := fragWriter.ArgWriter(f.isArg2Fragmented)
+	if err != nil {
+		return fmt.Errorf("get arg2 writer: %v", err)
+	}
+
+	if err := writeArg2WithAppends(arg2Writer, f.arg2(), f.arg2Appends); err != nil {
+		return fmt.Errorf("write arg2: %v", err)
+	}
+	if f.isArg2Fragmented {
+		return nil
+	}
+
+	if err := NewArgWriter(fragWriter.ArgWriter(true)).Write(f.arg3()); err != nil {
+		return errors.New("arg3 write failed")
+	}
```

Comment:
does using `NewArgWriter` have a performance overhead / allocs? It's a little bit more verbose to do it without the arg writer so want to understand if there's a performance impact here

Reply:
```
$ cat withappends.log | grep allocs
   10214	    114906 ns/op	   8.91 MB/s	    1025 B/op	      23 allocs/op
$ cat withappends-noargwriterhelper.log | grep allocs
   10196	    122726 ns/op	   8.34 MB/s	    1025 B/op	      23 allocs/op
```
seems there isn't much of a difference...